pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -243,10 +248,62 @@ stdenv.mkDerivation (finalAttrs: {
     ln -s $out/bin/qemu-system-${stdenv.hostPlatform.qemuArch} $out/bin/qemu-kvm
   '';
 
+  # Normal qemu package does not have installcheck make target.
+  # toolsOnly has custom check.
+  doInstallCheck = toolsOnly;
+
+  installCheckPhase =
+    if toolsOnly
+    then ''
+      runHook preInstallCheck
+
+      # Check the bin directory for unexpected files.
+      (
+        cd $out/bin;
+        echo binaries in qemu-utils package:
+        ls -al
+        for bin in *; do
+          case $bin in
+            elf2dmp) ;;
+            qemu-edid) ;;
+            qemu-img) ;;
+            qemu-io) ;;
+            qemu-nbd) ;;
+            qemu-pr-helper) ;;
+            qemu-storage-daemon) ;;
+            qemu-kvm)
+              if test -L $bin && ! test -e $bin; then
+                echo "removing broken $bin symlink"
+                echo "upstream build system should not write it, but it does, and it's not a big deal"
+                rm $bin
```

Comment:
This seems like a very weird thing to be doing in installCheckPhase.

Reply:
I made a mistake. Thanks!